escrow/contracts/Escrow.cdc

```diff
@@ -98,26 +97,21 @@ pub contract Escrow {
         // Burns an NFT entry from the leaderboard.
         access(contract) fun burn(nftID: UInt64) {
            // Check to see if the entry exists.
-           pre {
-               self.entriesData[nftID] != nil : "Entry does not exist with this NFT ID"
-           }
-
-           self.entriesData.remove(key: nftID)!
-           let token <- self.collection.withdraw(withdrawID: nftID)
-           emit EntryBurned(leaderboardName: self.name, nftID: nftID)
+            pre {
+                self.entriesData[nftID] != nil : "Entry does not exist with this NFT ID"
+            }
 
-           // Decrement entries length.
-           self.entriesLength = self.entriesLength - 1
+            self.entriesData.remove(key: nftID)!
+            let token <- self.collection.withdraw(withdrawID: nftID)
+            emit EntryBurned(leaderboardName: self.name, nftID: nftID)
 
-           destroy token
-        }
+            // Decrement entries length.
+            self.entriesLength = self.entriesLength - 1
 
-        // Destructor for Leaderboard resource.
-        destroy() {
-            destroy self.collection
+            destroy token
```

Comment:
where's token from?

Reply:
it's withdrawn from the Leaderboard's `collection` field

Reply:
oh sorry, i read it wrong. I thought this was inside the destructor